application/backend/src/main/resources/apps/typerefinery/components/structure/page/clientlibs/functions.js

```diff
@@ -81,16 +78,37 @@ window.MessageService.Client = MessageService.Client || {};
             }
         );
     }
+
+    ns.toggleTheme = () => {
+        const themeStyles = document.getElementById("themeStyles");
+        if (themeStyles) {
+            const currentTheme = themeStyles.getAttribute('active') || 'dark';
+            const newTheme = currentTheme === 'light' ? 'dark' : 'light';
+            themeStyles.setAttribute("active", newTheme)
+            themeStyles.setAttribute("href", `/apps/typerefinery/web_root/${newTheme}.css`)
+            localStorage.setItem("theme", newTheme);
+            setTimeout(() => {
+                Object.entries(chartInstances).forEach($chart => {
+                    $chart[1]?.update();
+                })
+            }, 15);
+        }
+    }
+
     ns.init = () => {
         const rootElement = document.querySelector(':root');
         themeNs.rootElementStyle = getComputedStyle(rootElement);
         ns.handleTheme();
 
+        $("#themeHandler").click(function () {
```

Comment:
is themeHandler a namespace? you are searchin for all #themeHandler in whole page? eveything within init is local to a component!

Reply:
Here we are just adding the click event listener to that id.